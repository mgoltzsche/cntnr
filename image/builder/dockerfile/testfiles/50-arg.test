FROM alpine:3.7

ARG argd entrypoint.sh
ARG argp
ARG argusr 0:0
ARG argimg alpine:3.7
ARG argvol /data
ARG port1 80
ARG port2 443/tcp

ENV MYVARX ${argd}
ENV MYVARY ${argp}

COPY $argd /dest
COPY /entrypoint.sh $argd
COPY --chown=$argusr /entrypoint.sh /dest
COPY --from=$argimg etc/hosts /dest

EXPOSE $port1 $port2

LABEL $argd=$argp

ARG argstop SIGTERM
STOPSIGNAL $argstop

USER $argusr

VOLUME $argp $argvol

WORKDIR $argvol
WORKDIR /

RUN echo $argd

# Should not substitute
ENTRYPOINT echo $argd
ENTRYPOINT [ "/bin/echo", "$argd" ]

# ASSERT CFG /config/StopSignal=SIGTERM