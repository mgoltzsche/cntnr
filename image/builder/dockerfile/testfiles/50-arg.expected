FROM alpine:3.7
NAME 0

ENV "MYVARX"="entrypoint.sh"
ENV "MYVARY"="pval"

COPY dir="./ctx" "entrypoint.sh" "/dest" nil
COPY dir="./ctx" "/entrypoint.sh" "entrypoint.sh" nil
COPY dir="./ctx" "/entrypoint.sh" "/dest" 0:0
COPY image="alpine:3.7" "etc/hosts" "/dest" nil

EXPOSE 80 443/tcp

LABEL "entrypoint.sh"="pval"

STOPSIGNAL SIGTERM

USER 0:0

VOLUME "pval" "/data"

WORKDIR /data
WORKDIR /

RUN "argd"="entrypoint.sh" "argimg"="alpine:3.7" "argp"="pval" "argstop"="SIGTERM" "argusr"="0:0" "argvol"="/data" "port1"="80" "port2"="443/tcp""/bin/sh" "-c" "echo $argd"

ENTRYPOINT "/bin/sh" "-c" "echo $argd"
ENTRYPOINT "/bin/echo" "$argd"